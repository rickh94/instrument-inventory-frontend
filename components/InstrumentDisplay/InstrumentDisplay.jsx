import React, { useState } from 'react'
import PropTypes from 'prop-types'

import {
  Grid,
  List,
  ListItem,
  ListItemText,
  makeStyles,
  Modal,
} from '@material-ui/core'
import ExifOrientationImg from 'react-exif-orientation-img'

import { LoadingHeader } from '..'
import { stars, yesOrNo } from '../../libs/display'

const useStyles = makeStyles({
  thumbnail: {
    marginLeft: 'auto',
    cursor: 'pointer',
  },
  fullImage: {
    position: 'absolute',
    left: '15vw',
    top: '10vh',
    width: '75vw',
    imageOrientation: 'from-image',
    border: 'none',
  },
})

const InstrumentDisplay = ({ isLoading, schema, body, omitFields = [] }) => {
  const [showPhoto, setShowPhoto] = useState(false)
  const classes = useStyles()
  return (
    <React.Fragment>
      <Grid container direction="row">
        <LoadingHeader isLoading={isLoading} title={`${body.type} ${body.number}`} />
        {body.photoUrls && (
          <Grid item className={classes.thumbnail} onClick={() => setShowPhoto(true)}>
            <img src={body.photoUrls.thumbnail} width="50px" height="50px" />
          </Grid>
        )}
      </Grid>
      <List disablePadding component="ul">
        <InstrumentDisplayFields
          schema={schema}
          body={body}
          omitFields={[...omitFields, 'id', 'type', 'number', 'photoUrls']}
        />
      </List>
      {body.photoUrls && (
        <Modal open={showPhoto} onClose={() => setShowPhoto(false)}>
          <ExifOrientationImg
            src={body.photoUrls.full}
            className={classes.fullImage}
            alt="instrument photo"
          />
        </Modal>
      )}
    </React.Fragment>
  )
}

InstrumentDisplay.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  schema: PropTypes.object,
  body: PropTypes.object,
  omitFields: PropTypes.array,
}

const InstrumentDisplayFields = ({ schema, body, omitFields }) => (
  <List disablePadding component="ul">
    {Object.keys(schema.properties)
      .filter(field => (omitFields.includes(field) ? null : field))
      .map(field => (
        <InfoItem
          title={schema.properties[field].title}
          data={body[field]}
          type={schema.properties[field].type}
          key={field}
        />
      ))}
  </List>
)

InstrumentDisplayFields.propTypes = {
  schema: PropTypes.object,
  body: PropTypes.object,
  omitFields: PropTypes.array,
}

const InfoItem = ({ title, data, type }) => {
  let text
  switch (type) {
    case 'array':
      text = data ? data.join(', ') : ''
      break
    case 'integer':
      text = stars(data)
      break
    case 'boolean':
      text = yesOrNo(data)
      break
    default:
      text = data
  }

  return (
    <ListItem component="li">
      <ListItemText
        primary={title}
        secondary={text}
        primaryTypographyProps={{ variant: 'subtitle1', color: 'textSecondary' }}
        secondaryTypographyProps={{ variant: 'subtitle2', color: 'textPrimary' }}
      />
    </ListItem>
  )
}

InfoItem.propTypes = {
  title: PropTypes.string.isRequired,
  data: PropTypes.any,
  type: PropTypes.string.isRequired,
}

export default InstrumentDisplay
