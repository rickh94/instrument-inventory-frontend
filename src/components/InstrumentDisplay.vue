<template>
  <v-modal v-if="currentInstrument"
           @close="editing = false; clearCurrentInstrument()"
           :width-class="editing ? 'sm:max-w-xl' : 'sm:max-w-lg'">
    <div class="rounded w-full sm:max-w-2xl" v-if="editing">
      <v-instrument-form
        @editSuccess="editing = false"
        @cancel="editing = false"
        class="max-h-screen overflow-auto overflow-x-hidden rounded"></v-instrument-form>
    </div>
    <div v-else class="w-full">
      <div class="flex justify-between mb-4">
        <h6 class="text-xl font-bold text-gray-900">
          <span v-if="currentInstrument.size !== 'N/A'">
           {{ currentInstrument.size }}
          </span>
          {{ currentInstrument.type }} {{
            currentInstrument.number }}
        </h6>
        <button class="appearance-none text-white bg-red-600 px-1 rounded shadow hover:bg-red-800 hover:shadow-lg text-sm"
                @click="clearCurrentInstrument" title="Close Instrument Display">
          <svg xmlns="http://www.w3.org/2000/svg" class="h-5 w-5" viewBox="0 0 20 20" fill="currentColor">
            <path fill-rule="evenodd"
                  d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
                  clip-rule="evenodd" />
          </svg>
        </button>
      </div>
      <div>
        <span class="font-bold text-gray-700 mb-2">Size: </span>{{ currentInstrument.size }}
      </div>
      <div>
        <span class="font-bold text-gray-700 mb-2">Location: </span>{{ currentInstrument.location }}
      </div>
      <div>
        <span class="font-bold text-gray-700 mb-2">Assigned To: </span>{{ currentInstrument.assignedTo }}
      </div>
      <div>
        <span class="font-bold text-gray-700 mb-2">Notes: </span>{{
          currentInstrument.conditionNotes }}
      </div>
      <div>
        <span class="font-bold text-gray-700 mb-2">Condition: </span>{{ currentInstrument.condition }}
      </div>
      <div>
        <span class="font-bold text-gray-700 mb-2">Quality: </span>{{ currentInstrument.quality }}
      </div>
      <div>
        <span class="font-bold text-gray-700 mb-2">Gifted To Student: </span>{{ currentInstrument.gifted ?
        "Yes" : "No" }}
      </div>
      <div v-if="currentInstrument.history">
        <span class="font-bold text-gray-700 mb-2">History: </span>{{ currentInstrument.history.join(", ")
        }}
      </div>
      <div class="font-bold text-gray-700" v-else>
        No Instrument History
      </div>
      <div>
        <span class="font-bold text-gray-700 mb-2">Archived</span> {{ currentInstrument.archived ? "Yes" : "No" }}
      </div>
      <v-loading-buttons loader-class="w-80 mr-2" :loading="loading" container-class="mt-2 grid grid-cols-2 sm:grid-cols-4 gap-y-2">
        <v-edit-button v-if="isAdmin" @click="edit" center />
        <router-link
          v-if="$route.path !== '/sign-out'"
          to="/sign-out"
          class="inline-flex items-center justify-center font-bold mx-1 appearance-none bg-blue-600 text-white px-3 py-1 shadow rounded hover:bg-blue-800 hover:shadow-lg">
          <span class="inline-flex items-center">
            <svg xmlns="http://www.w3.org/2000/svg" class="h-5 w-5 mr-1" viewBox="0 0 20 20" fill="currentColor"> <path d="M9 2a1 1 0 000 2h2a1 1 0 100-2H9z" />
              <path fill-rule="evenodd"
                    d="M4 5a2 2 0 012-2 3 3 0 003 3h2a3 3 0 003-3 2 2 0 012 2v11a2 2 0 01-2 2H6a2 2 0 01-2-2V5zm3 4a1 1 0 000 2h.01a1 1 0 100-2H7zm3 0a1 1 0 000 2h3a1 1 0 100-2h-3zm-3 4a1 1 0 100 2h.01a1 1 0 100-2H7zm3 0a1 1 0 100 2h3a1 1 0 100-2h-3z"
                    clip-rule="evenodd" />
            </svg>
            Assign
          </span>
        </router-link>
        <v-retrieve-button @click="retrieve" center />
        <button v-if="isAdmin" @click="toggleArchived"
                class="inline-flex items-center justify-center font-bold mx-1 appearance-none bg-pink-600 text-white px-4 py-1 shadow rounded hover:bg-pink-800 hover:shadow-lg">
          <span class="inline-flex items-center">
            <svg xmlns="http://www.w3.org/2000/svg"
                 class="h-6 w-6 mr-1"
                 fill="none"
                 viewBox="0 0 24 24"
                 stroke="currentColor"
                 v-if="currentInstrument.archived">
              <path stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    d="M5 8h14M5 8a2 2 0 110-4h14a2 2 0 110 4M5 8v10a2 2 0 002 2h10a2 2 0 002-2V8m-9 4h4" />
            </svg>
            <svg xmlns="http://www.w3.org/2000/svg" class="h-5 w-5 mr-1" viewBox="0 0 20 20" fill="currentColor" v-else>
              <path d="M4 3a2 2 0 100 4h12a2 2 0 100-4H4z" />
              <path fill-rule="evenodd"
                    d="M3 8h14v7a2 2 0 01-2 2H5a2 2 0 01-2-2V8zm5 3a1 1 0 011-1h2a1 1 0 110 2H9a1 1 0 01-1-1z"
                    clip-rule="evenodd" />
            </svg>
            {{ currentInstrument.archived ? "Un-archive" : "Archive" }}
          </span>
        </button>
      </v-loading-buttons>
<!--      <div class="mt-2 grid grid-cols-2 sm:grid-cols-4 gap-y-2" v-else>-->
<!--      </div>-->
    </div>
  </v-modal>
</template>
<script>
import { mapMutations, mapState } from "vuex";
import VModal from "@/components/UI/VModal";
import { API } from "aws-amplify";
import checkAdmin from "@/mixins/checkAdmin";
import VRetrieveButton from "@/components/UI/buttons/VRetrieveButton";
import VEditButton from "@/components/UI/buttons/VEditButton";
import VLoadingButtons from "@/components/UI/VLoadingButtons";

export default {
  name: "InstrumentDisplay",
  mixins: [checkAdmin],
  data() {
    return {
      editing: false,
      loading: false,
    };
  },
  components: {
    VLoadingButtons,
    VEditButton,
    VRetrieveButton,
    VInstrumentForm: () => import("@/components/createComponents/VInstrumentForm"),
    VModal,
  },
  methods: {
    ...mapMutations(["clearCurrentInstrument", "setCurrentInstrument", "updateCurrentInstrument", "clearNewInstrumentNumber"]),
    async toggleArchived() {
      const wasArchived = this.currentInstrument.archived;
      try {
        this.loading = true;
        const response = await API.put("instrument-inventory", `instruments/${this.currentInstrument.id}`, {
          body: {
            ...this.currentInstrument,
            archived: !this.currentInstrument.archived,
          },
        });
        this.loading = false;
        if (wasArchived) {
          this.$toasted.info(`Instrument ${response.item.number} un-archived`, { duration: 2000 });
        } else {
          this.$toasted.info(`Instrument ${response.item.number} archived`, { duration: 2000 });
        }
        this.updateCurrentInstrument(response.item);
      } catch (e) {
        this.loading = false;
        this.$toasted.error(`Error: ${e.response.data}`, { duration: 2000 });
      }
    },
    async retrieve() {
      this.loading = true;
      try {
        const response = await API.post("instrument-inventory", "retrieve-single",
          { body: { number: this.currentInstrument.number } });
        console.log(response);
        this.updateCurrentInstrument(response.item);
        this.$toasted.info(response.message, { duration: 2000 });
        this.loading = false;
      } catch (e) {
        this.$toasted.error(e.response.data, { duration: 2000 });
        console.log(e);
        this.loading = false;
      }

    },
    async handleEditSuccess() {
      this.editing = false;
      this.$emit("instrumentUpdated", this.currentInstrument.id);
    },
    edit() {
      this.clearNewInstrumentNumber();
      this.editing = true;
    },
  },
  computed: mapState(["currentInstrument"]),
};
</script>
