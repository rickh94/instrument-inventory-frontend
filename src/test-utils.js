/* eslint-disable react/prop-types */
import React from 'react'
import { MuiThemeProvider, createMuiTheme, indigo } from '@material-ui/core'
import { HelpersContext, SchemaContext } from './contexts'
import { MemoryRouter } from 'react-router-dom'
import { render } from '@testing-library/react'

const theme = createMuiTheme({
  palette: {
    primary: indigo,
    secondary: {
      main: '#0A81D1',
    },
  },
})

export const showAlert = jest.fn()
export const setSearchResults = jest.fn()
export const setFilterResults = jest.fn()
export const searchResults = jest.fn()
export const filterResults = jest.fn()

export const TestTheme = ({ children }) => (
  <MuiThemeProvider theme={theme}>{children}</MuiThemeProvider>
)

export const TestHelpers = ({ children }) => (
  <HelpersContext.Provider
    value={{
      showAlert,
      setSearchResults,
      setFilterResults,
      searchResults,
      filterResults,
    }}
  >
    {children}
  </HelpersContext.Provider>
)

export const TestRouter = ({ children }) => <MemoryRouter>{children}</MemoryRouter>

export const TestSchema = ({ children, schema = { components: { schemas: {} } } }) => (
  <SchemaContext.Provider value={schema}>{children}</SchemaContext.Provider>
)

export const TestEverything = ({ children }) => (
  <TestRouter>
    <TestTheme>
      <TestHelpers>{children}</TestHelpers>
    </TestTheme>
  </TestRouter>
)

// eslint-disable-next-line no-undef
export const flushPromises = () => new Promise(setImmediate)

const customRender = (ui, options, schema = { components: { schemas: {} } }) =>
  render(<TestSchema schema={schema}>{ui}</TestSchema>, {
    wrapper: TestEverything,
    ...options,
  })

export * from '@testing-library/react'

export { customRender as render }
