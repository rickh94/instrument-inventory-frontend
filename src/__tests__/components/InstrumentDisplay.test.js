import React from 'react'
import InstrumentDisplay, {
  InfoItem,
  InstrumentDisplayFields,
} from '../../components/InstrumentDisplay'
import { render, cleanup } from '../../test-utils'

const instrumentSchema = {
  properties: {
    assignedTo: {
      title: 'Assigned To',
      type: 'string',
    },
    condition: {
      title: 'Condition',
      type: 'integer',
    },
    conditionNotes: {
      title: 'Condition Notes',
      type: 'string',
    },
    gifted: {
      title: 'Gifted To Student',
      type: 'boolean',
    },
    location: {
      title: 'Instrument Location',
      type: 'string',
    },
    maintenanceNotes: {
      title: 'Maintenance Notes',
      type: 'string',
    },
    number: {
      title: 'Instrument Number',
      type: 'string',
    },
    quality: {
      title: 'Quality',
      type: 'integer',
    },
    size: {
      title: 'Instrument Size',
      type: 'string',
    },
    type: {
      title: 'Instrument Type',
      type: 'string',
    },
  },
}
const body = {
  type: 'Violin',
  number: '1-808',
  size: '4/4',
  location: 'Office',
  assignedTo: 'Some Name',
  condition: 5,
  quality: 3,
  conditionNotes: 'testconditionnotes',
  maintenanceNotes: 'testmaintenancenotes',
  gifted: true,
}

afterEach(cleanup)

describe('Instrument Display components', () => {
  describe('InstrumentDisplay', () => {
    it('matches snapshot', () => {
      const { container } = render(
        <InstrumentDisplay schema={instrumentSchema} body={body} isLoading={false} />
      )
      expect(container).toMatchSnapshot()
    })

    it('renders data from props', () => {
      const { getAllByText } = render(
        <InstrumentDisplay schema={instrumentSchema} body={body} isLoading={false} />
      )

      expect(getAllByText('Violin 1-808').length).not.toBe(0)
      expect(getAllByText(body.size).length).not.toBe(0)
      expect(getAllByText(body.location).length).not.toBe(0)
      expect(getAllByText(body.assignedTo).length).not.toBe(0)
      expect(getAllByText('★'.repeat(body.condition)).length).not.toBe(0)
      expect(getAllByText('★'.repeat(body.quality)).length).not.toBe(0)
      expect(getAllByText(body.conditionNotes).length).not.toBe(0)
      expect(getAllByText(body.maintenanceNotes).length).not.toBe(0)
    })
  })

  describe('InstrumentDisplayFields', () => {
    it('matches snapshot', () => {
      const { container } = render(
        <InstrumentDisplayFields schema={instrumentSchema} body={body} />
      )
      expect(container).toMatchSnapshot()
    })

    it('renders data from props', () => {
      const { queryAllByText } = render(
        <InstrumentDisplayFields schema={instrumentSchema} body={body} />
      )

      expect(queryAllByText('4/4').length).not.toBe(0)
      expect(queryAllByText('Assigned To').length).not.toBe(0)
    })

    it('does not render omitted fields', () => {
      const { queryByText } = render(
        <InstrumentDisplayFields
          schema={instrumentSchema}
          body={body}
          omitFields={['number', 'type']}
        />
      )

      expect(queryByText('Instrument Number')).toBeFalsy()
      expect(queryByText('1-808')).toBeFalsy()
    })
  })

  describe('InfoItem', () => {
    it('matches snapshot', () => {
      const { container } = render(
        <InfoItem title="test title" data="title" type="string" />
      )
      expect(container).toMatchSnapshot()
    })

    it('renders strings correctly', () => {
      const { getAllByText } = render(
        <InfoItem title="Test Title" data="Something" type="string" />
      )
      expect(getAllByText('Test Title')).toBeTruthy()
      expect(getAllByText('Something')).toBeTruthy()
    })

    it('renders arrays correctly', () => {
      const { getAllByText } = render(
        <InfoItem title="Array" data={['one', 'two', 'three']} type="array" />
      )
      expect(getAllByText('one, two, three')).toBeTruthy()
    })

    it('renders integers as stars', () => {
      const { getAllByText } = render(
        <InfoItem title="Stars" data={3} type="integer" />
      )
      expect(getAllByText('★'.repeat(3))).toBeTruthy()
    })

    it('renders true boolean as yes', () => {
      const { getAllByText, queryByText } = render(
        <InfoItem title="True" data={true} type="boolean" />
      )
      expect(getAllByText('Yes')).toBeTruthy()
      expect(queryByText('No')).toBeFalsy()
    })
  })

  describe('InfoItem false', () => {
    it('renders false boolean as no', () => {
      const { getAllByText, queryByText } = render(
        <InfoItem title="False" data={false} type="boolean" />
      )
      expect(getAllByText('No')).toBeTruthy()
      expect(queryByText('Yes')).toBeFalsy()
    })
  })
})
