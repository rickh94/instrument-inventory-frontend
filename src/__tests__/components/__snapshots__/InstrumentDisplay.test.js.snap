// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instrument Display components InfoItem matches snapshot 1`] = `
<div>
  <li
    class="MuiListItem-root MuiListItem-gutters"
  >
    <div
      class="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
      >
        test title
      </span>
      <h6
        class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
      >
        title
      </h6>
    </div>
  </li>
</div>
`;

exports[`Instrument Display components InstrumentDisplay matches snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container"
  >
    <div
      class="makeStyles-heading-106"
      data-testid="loading-header"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit"
      >
        Violin 1-808
      </h5>
    </div>
  </div>
  <ul
    class="MuiList-root"
  >
    <ul
      class="MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
          >
            Assigned To
          </span>
          <h6
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
          >
            Some Name
          </h6>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
          >
            Condition
          </span>
          <h6
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
          >
            ★★★★★
          </h6>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
          >
            Condition Notes
          </span>
          <h6
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
          >
            testconditionnotes
          </h6>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
          >
            Gifted To Student
          </span>
          <h6
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
          >
            Yes
          </h6>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
          >
            Instrument Location
          </span>
          <h6
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
          >
            Office
          </h6>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
          >
            Maintenance Notes
          </span>
          <h6
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
          >
            testmaintenancenotes
          </h6>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
          >
            Quality
          </span>
          <h6
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
          >
            ★★★
          </h6>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
          >
            Instrument Size
          </span>
          <h6
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
          >
            4/4
          </h6>
        </div>
      </li>
    </ul>
  </ul>
</div>
`;

exports[`Instrument Display components InstrumentDisplayFields matches snapshot 1`] = `
<div>
  <ul
    class="MuiList-root"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
        >
          Assigned To
        </span>
        <h6
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
        >
          Some Name
        </h6>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
        >
          Condition
        </span>
        <h6
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
        >
          ★★★★★
        </h6>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
        >
          Condition Notes
        </span>
        <h6
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
        >
          testconditionnotes
        </h6>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
        >
          Gifted To Student
        </span>
        <h6
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
        >
          Yes
        </h6>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
        >
          Instrument Location
        </span>
        <h6
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
        >
          Office
        </h6>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
        >
          Maintenance Notes
        </span>
        <h6
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
        >
          testmaintenancenotes
        </h6>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
        >
          Instrument Number
        </span>
        <h6
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
        >
          1-808
        </h6>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
        >
          Quality
        </span>
        <h6
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
        >
          ★★★
        </h6>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
        >
          Instrument Size
        </span>
        <h6
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
        >
          4/4
        </h6>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
        >
          Instrument Type
        </span>
        <h6
          class="MuiTypography-root MuiListItemText-secondary MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
        >
          Violin
        </h6>
      </div>
    </li>
  </ul>
</div>
`;
